### PR TITLE
Classes view loading improvements

### PR DESCRIPTION
**Pull request by differentmatt**
Skip sessions for archived classrooms
Render UI before all sessions loaded
Only create and insert course nag subview once

Artificially threw random 503s for member-sessions calls to ensure it
works alright.  Course bubbles with 503s remain in Loading.. state.

### CODE REVIEW

File: app/views/courses/TeacherClassesView.coffee [hunk 1]
```diff
@@ -97,10 +97,14 @@ module.exports = class TeacherClassesView extends RootView
...
         classroom.sessions = new LevelSessions()
-        jqxhrs = classroom.sessions.fetchForAllClassroomMembers(classroom)
-        if jqxhrs.length > 0
-          @supermodel.trackRequests(jqxhrs)
+        Promise.all(classroom.sessions.fetchForAllClassroomMembers(classroom))
+        .then (results) =>
+          return if @destroyed
+          helper.calculateDots(@classrooms, @courses, @courseInstances)
+          @calculateQuestCompletion()
+          @render()
```
Comment (sderickson) [inline app/views/courses/TeacherClassesView.coffee]: Did you test this if one fails?

File: app/views/courses/TeacherClassesView.coffee [hunk 2]
```diff
@@ -118,8 +122,9 @@ module.exports = class TeacherClassesView extends RootView
 
   afterRender: ->
     super()
-    @courseNagSubview = new CourseNagSubview()
-    @insertSubView(@courseNagSubview)
+    unless @courseNagSubview
+      @courseNagSubview = new CourseNagSubview()
+      @insertSubView(@courseNagSubview)
```
Comment (sderickson) [inline app/views/courses/TeacherClassesView.coffee]: I think the `insertSubView` line will need to happen after each render.

Comment (differentmatt) [reply app/views/courses/TeacherClassesView.coffee]: Just insert without new doesn't work, and doing this on every render generates a bunch of duplicate prepaid fetches.  This seems to work fine, teacher quest stuff gets updated as data loads.

File: app/views/courses/TeacherClassesView.coffee (hunk 1 above)
Comment (differentmatt) [reply app/views/courses/TeacherClassesView.coffee]: Yes, it leaves the class in a loading state with all the course bubbles grey, hover popup says Loading..

**Pull request merged**